### ISSUE TITLE
feat(linter): implement typescript/prefer-nullish-coalescing rule

### ISSUE THREAD

**Issue by higumachan** [oxc-project/oxc#12410]
## Description

The `@typescript-eslint/prefer-nullish-coalescing` rule enforces using the nullish coalescing operator (`??`) instead of logical chaining (`||`) when providing default values. This helps avoid bugs where falsy values like `0`, `""`, or `false` are unintentionally replaced.

## Background

This rule is part of the `@typescript-eslint/stylistic-type-checked` configuration and requires type information. Currently, oxc filters out type-checked rules in the tracking system (see `tasks/lint_rules/src/eslint-rules.mjs` lines 94-102).

## Rule Details

- **Name**: `prefer-nullish-coalescing`
- **Documentation**: https://typescript-eslint.io/rules/prefer-nullish-coalescing/
- **Category**: Stylistic type-checked
- **Requires**: TypeScript 3.7+
- **Fixable**: Yes (provides editor suggestions)

### Examples

**❌ Incorrect:**
```typescript
const foo = bar || 'default';
const baz = (qux === null || qux === undefined) ? 'default' : qux;
```

**✅ Correct:**
```typescript
const foo = bar ?? 'default';
const baz = qux ?? 'default';
```

## Configuration Options

- `ignoreTernaryTests`: Ignore ternary expressions that could be simplified
- `ignoreConditionalTests`: Ignore cases within conditional tests (default: true)
- `ignoreMixedLogicalExpressions`: Ignore logical OR within mixed logical expressions
- `ignorePrimitives`: Configure handling of specific primitive types
- `allowRuleToRunWithoutStrictNullChecksIKnowWhatIAmDoing`: Allow running without `strictNullChecks`

## Implementation Considerations

Since this is a type-checked rule, implementing it in oxc would require:
1. Type information support in the linter
2. Ability to determine if a value can be nullish
3. Understanding of TypeScript's type system for accurate suggestions

## Related Work

- oxc already supports the nullish coalescing operator in the transformer (#4884)
- The minifier can constant fold nullish coalescing operators (#5761)
- Related rule `typescript/no-non-null-asserted-nullish-coalescing` is already implemented

## Priority

This is a valuable rule for TypeScript codebases as it helps prevent common bugs related to falsy value handling and promotes more explicit null/undefined checks.

**Comment by camc314** [comment]
type aware linting POC is currently being explored

**Comment by chenglou** [comment]
Can this be reopened now?

**Comment by camc314** [comment]
tracked here [https://github.com/oxc-project/tsgolint/issues/198](https://github.com/oxc-project/tsgolint/issues/198)
